api/src/test/java/org/apache/iceberg/expressions/TestIDReference.java

```diff
@@ -0,0 +1,132 @@
+    assertThat(ref1.name()).isNotEqualTo(ref3.name());
+
+    // Same name, different fieldId
+    assertThat(ref1).isNotEqualTo(ref4);
+  }
+
+  @Test
+  public void testIDReferenceBind() {
+    IDReference<Integer> ref = new IDReference<>("a", 34);
+    BoundReference<Integer> bound = ref.bind(SCHEMA.asStruct(), true);
+
+    assertThat(bound).isInstanceOf(BoundReference.class);
+    assertThat(bound.fieldId()).isEqualTo(34);
+    assertThat(bound.name()).isEqualTo("a");
+    assertThat(bound.type()).isEqualTo(Types.IntegerType.get());
+  }
+
+  @Test
+  public void testIDReferenceBindIgnoresCaseSensitivity() {
+    IDReference<Integer> ref = new IDReference<>("A", 34);
+
+    // Should work regardless of case sensitivity since we use fieldId
+    BoundReference<Integer> bound1 = ref.bind(SCHEMA.asStruct(), true);
+    BoundReference<Integer> bound2 = ref.bind(SCHEMA.asStruct(), false);
+
+    assertThat(bound1).isInstanceOf(BoundReference.class);
+    assertThat(bound2).isInstanceOf(BoundReference.class);
+    assertThat(bound1.fieldId()).isEqualTo(34);
+    assertThat(bound2.fieldId()).isEqualTo(34);
+  }
+
+  @Test
+  public void testIDReferenceBindWithInvalidId() {
+    IDReference<Integer> ref = new IDReference<>("invalid", 999);
+
+    assertThatThrownBy(() -> ref.bind(SCHEMA.asStruct(), true))
+        .isInstanceOf(ValidationException.class)
+        .hasMessageContaining(
+            "Cannot find field by id 999 in struct: struct<34: a: optional int, 35: s: required string>");
+  }
+
+  @Test
+  public void testIDReferenceRef() {
+    IDReference<Integer> ref = new IDReference<>("a", 34);
+    NamedReference<?> namedRef = ref.ref();
+
+    assertThat(namedRef.name()).isEqualTo("a");
+  }
+
+  @Test
+  public void testResolvedReferenceToString() {
+    IDReference<Integer> ref = new IDReference<>("a", 34);
+
+    assertThat(ref.toString()).isEqualTo("ref(name=\"a\", id=\"34\")");
+  }
+
+  @Test
+  public void testIDReferenceExpressionIntegration() {
+    // Test that IDReference works in expression predicates
+    UnboundPredicate<?> expr = Expressions.equal(Expressions.ref("a", 34), 5);
+    assertThat(expr).isInstanceOf(UnboundPredicate.class);
+
+    assertThat(expr.term()).isInstanceOf(IDReference.class);
+
+    IDReference<?> resolvedRef = (IDReference<?>) expr.term();
+    assertThat(resolvedRef.name()).isEqualTo("a");
+    assertThat(resolvedRef.id()).isEqualTo(34);
+  }
+
+  @Test
+  public void testIDReferenceUnbind() {
+    // Test that unbinding a bound reference returns a NamedReference for compatibility
+    Expression expr = Expressions.equal(Expressions.ref("a", 34), 5);
+    Expression boundExpr = Binder.bind(SCHEMA.asStruct(), expr, true);
+
+    assertThat(boundExpr).isInstanceOf(BoundPredicate.class);
+    BoundPredicate<?> boundPred = (BoundPredicate<?>) boundExpr;
+
+    UnboundTerm<?> unbound = ExpressionUtil.unbind(boundPred.term());
+    assertThat(unbound).isInstanceOf(NamedReference.class);
```

Comment:
We discussed this previously. Now with `IDReference` extends from `NamedReference`, we can change the `ExpressionUtil.unbind` implementation to always return an `IDReference` object without change the API return type (stay as `NamedReference`).

Reply:
i mostly deferred this change due this feedback https://github.com/singhpk234/iceberg/pull/270#discussion_r2453198673, please let me know if you feel strongly about it.